Source/cmNinjaNormalTargetGenerator.cxx

```diff
@@ -927,6 +946,25 @@ void cmNinjaNormalTargetGenerator::WriteNvidiaDeviceLinkStatement(
   this->WriteNvidiaDeviceLinkRule(usedResponseFile, config);
 }
 
+void cmNinjaNormalTargetGenerator::CreateLinkCommand(
+  std::vector<std::string> commands, const cmNinjaBuild& linkBuild,
+  std::vector<std::string>& librariesPaths, std::set<int> ranlibIndexes)
+{
+  std::string version =
+    cmToCStrSafe(this->Makefile->GetDefinition("CMAKE_EXPORT_LINK_COMMANDS"));
+  if (!version.empty()) {
+    for (size_t i = 0; i < commands.size(); ++i) {
+      // skip ranlib command
+      if (ranlibIndexes.find(i) == ranlibIndexes.end()) {
+        if (version == "0.0.1") {
+          this->GetGlobalGenerator()->AddLinkCommand(
```

Comment:
You use here AddLinkCommand, but in another place you use CreateLinkScriptJson. Is that ok?